core/src/main/java/com/linecorp/armeria/server/DefaultRoute.java

```diff
@@ -312,6 +310,66 @@ public boolean isCacheable() {
         return paramPredicates.isEmpty() && headerPredicates.isEmpty() && excludedRoutes().isEmpty();
     }
 
+    @Override
+    public boolean hasDuplicateRouteCondition(Route other) {
+        if (complexity != other.complexity()) {
+            return false;
+        }
+
+        if (getClass() != other.getClass()) {
+            return false;
+        }
+
+        if (!pathType().hasTriePath() || !other.pathType().hasTriePath()) {
+            return false;
+        }
+
+        if (!paths().get(1).equals(other.paths().get(1))) {
+            // Just compare trie path.
+            return false;
+        }
+
+        if (methods.stream().noneMatch(method -> other.methods().contains(method))) {
+            // No overlap in supported methods.
+            return false;
+        }
+        if (!consumes.isEmpty() &&
```

Comment:
Q) If `consumes` is empty, it can consume any media type. In the following situation, should we assume that the two `consumes` are duplicate? 

- this.consumes = [JSON]
- others.consumes = []

Reply:
That's a good point. I think that it was a bug, and we should trea them as the same.
I'll address this in a separate PR because it might introduce unexpected behavior changes.